x/nexus/keeper/wasmer_engine.go

```diff
@@ -0,0 +1,113 @@
+package keeper
+
+import (
+	wasmkeeper "github.com/CosmWasm/wasmd/x/wasm/keeper"
+	wasmtypes "github.com/CosmWasm/wasmd/x/wasm/types"
+	wasmvm "github.com/CosmWasm/wasmvm"
+	wasmvmtypes "github.com/CosmWasm/wasmvm/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/x/nexus/types"
+)
+
+// WasmerEngine is a wrapper around the WasmerEngine to add a transaction ID generator
+type WasmerEngine struct {
+	wasmtypes.WasmerEngine
+	txIDGenerator types.TxIDGenerator
+}
+
+// NewWasmerEngine wraps the given engine with a transaction ID generator
+func NewWasmerEngine(inner wasmtypes.WasmerEngine, txIDGenerator types.TxIDGenerator) wasmtypes.WasmerEngine {
+	return &WasmerEngine{WasmerEngine: inner, txIDGenerator: txIDGenerator}
+}
+
+func getCtx(querier wasmvm.Querier) sdk.Context {
+	return querier.(wasmkeeper.QueryHandler).Ctx
+}
+
+// Instantiate calls the inner engine and increments the transaction ID
+func (w *WasmerEngine) Instantiate(
+	checksum wasmvm.Checksum,
+	env wasmvmtypes.Env,
+	info wasmvmtypes.MessageInfo,
+	initMsg []byte,
+	store wasmvm.KVStore,
+	goapi wasmvm.GoAPI,
+	querier wasmvm.Querier,
+	gasMeter wasmvm.GasMeter,
+	gasLimit uint64,
+	deserCost wasmvmtypes.UFraction,
+) (*wasmvmtypes.Response, uint64, error) {
+	defer w.txIDGenerator.NextID(getCtx(querier))
```

Comment:
do we need to `defer`?

Reply:
I prefer having it so that it would start from `0`, if reset.